src/tasks/command/command_parser.c

```diff
@@ -0,0 +1,56 @@
+/**
+ * @author  Thomas Haile
+ * @date    2025-05-24
+ *
+ * Command parsing implementation
+ */
+
+#include "command_parser.h"
+#include "macros.h"
+
+/// @brief Parse packet and dispatch command to appropriate queue
+void dispatch_command(slate_t *slate, packet_t *packet)
+{
+    slate->number_commands_processed++;
+
+    Command command_id = (Command)packet->data[0];
+    LOG_INFO("Command ID Received: %i", command_id);
+
+    switch (command_id)
+    {
+        /* Payload Commands */
+        case TAKE_PHOTO:
+        case DOWNLOAD_PHOTO:
+        case TAKE_AND_SEND_PHOTO:

```

Comment:
It looks like you're directly passing through whatever serialized_command the command payload contains through to the task_data queue and (subsequently) into the RPi UART channel -- this is the expected behavior of the PAYLOAD_EXEC kind of command we discussed earlier.

Could we have this as just 1 PAYLOAD_EXEC command id?